core/templates/dev/head/pages/exploration_editor/simple_editor_tab/SimpleEditorManagerServiceSpec.js

```diff
@@ -0,0 +1,352 @@
+    expect(stubs.SimpleEditorShimService.saveAnswerGroups)
+      .toHaveBeenCalledWith(stateName, newAnswerGroups);
+    expect(questionList.getBindableQuestion).toHaveBeenCalledWith(stateName);
+  });
+
+  it('should save the default outcome', function() {
+    var stateName = 'Introduction';
+    var newDefaultOutcome = {
+      dest: 'Introduction',
+      feedback: [
+        '<p>One is wrong bcz ofisEmpty many reasons</p>'
+      ]
+    };
+    simpleEditorManagerService.tryToInit();
+    simpleEditorManagerService.saveDefaultOutcome(stateName,
+      newDefaultOutcome);
+    expect(stubs.SimpleEditorShimService.saveDefaultOutcome)
+      .toHaveBeenCalledWith(stateName, newDefaultOutcome);
+    expect(questionList.getBindableQuestion).toHaveBeenCalledWith(stateName);
+  });
+
+  it('should save the bridge html', function() {
+    var stateName = 'Introduction';
+    var newBridgeHtml = '<p> lets move to next Qsn </p>';
+    simpleEditorManagerService.tryToInit();
+    simpleEditorManagerService.saveBridgeHtml(stateName, newBridgeHtml);
+    expect(stubs.SimpleEditorShimService.saveBridgeHtml)
+      .toHaveBeenCalledWith(questionList.getNextStateName(stateName),
+        newBridgeHtml);
+    expect(questionList.getBindableQuestion).toHaveBeenCalledWith(stateName);
+  });
+
+  it('should add new question', function() {
+    var DEFAULT_INTERACTION = {
+      ID: 'MultipleChoiceInput',
+      CUSTOMIZATION_ARGS: {
+        choices: {
+          value: ['<p>Option 1</p>']
+        }
+      }
+    };
+
+    var lastStateName = (questionList.isEmpty() ? SimpleEditorShimService
+        .getInitStateName() : questionList.getLastQuestion()
+          .getDestinationStateName());
+
+    var defaultOutcome = {
+      dest: lastStateName,
+      feedback: [''],
+      param_changes: []
+    };
+
+    simpleEditorManagerService.tryToInit();
+    simpleEditorManagerService.addNewQuestion();
+
+    expect(stubs.SimpleEditorShimService.saveInteractionId)
+      .toHaveBeenCalledWith(lastStateName, DEFAULT_INTERACTION.ID);
+    expect(stubs.SimpleEditorShimService.saveCustomizationArgs)
+      .toHaveBeenCalledWith(lastStateName,
+        DEFAULT_INTERACTION.CUSTOMIZATION_ARGS);
+    expect(stubs.SimpleEditorShimService.saveDefaultOutcome)
+      .toHaveBeenCalledWith(lastStateName, defaultOutcome);
+    expect(questionList.addQuestion).toHaveBeenCalled();
+    expect(stubs.QuestionObjectFactory.create).toHaveBeenCalled();
+  });
+
+  it('should add state', function() {
+    simpleEditorManagerService.tryToInit();
+    simpleEditorManagerService.addState();
+    expect(stubs.SimpleEditorShimService.addState).toHaveBeenCalled();
+    expect(stubs.SimpleEditorShimService.saveInteractionId).toHaveBeenCalled();
+    expect(stubs.SimpleEditorShimService.saveCustomizationArgs)
+      .toHaveBeenCalled();
+    expect(stubs.SimpleEditorShimService.saveDefaultOutcome)
+      .toHaveBeenCalled();
+  });
```

Comment:
I am having difficulty implementing and designing tests required as mentioned above. can I get help form some one who has good experience writing Angular tests?